airbyte-integrations/connectors/source-google-ad-manager/source_google_ad_manager/streams.py

```diff
@@ -0,0 +1,324 @@
+        """generate the report query for the ad unit per referrer report
+
+        Returns:
+            _type_: _description_
+        """
+        targets_ids = [targeting_value['id'] for targeting_value in targeting_values if targeting_value["id"]]
+        targets_ids = ", ".join([str(target_id) for target_id in targets_ids])
+        report_job = {"reportQuery": {}}
+        custom_traffic_source_columns = [
+                "TOTAL_LINE_ITEM_LEVEL_CPM_AND_CPC_REVENUE",
+                "TOTAL_LINE_ITEM_LEVEL_IMPRESSIONS",
+                "TOTAL_LINE_ITEM_LEVEL_CLICKS",
+                "TOTAL_LINE_ITEM_LEVEL_WITHOUT_CPD_AVERAGE_ECPM",
+                "TOTAL_ACTIVE_VIEW_VIEWABLE_IMPRESSIONS",
+                "TOTAL_ACTIVE_VIEW_MEASURABLE_IMPRESSIONS",
+                "TOTAL_ACTIVE_VIEW_ELIGIBLE_IMPRESSIONS",
+            ]
+        dimensions = ['ADVERTISER_NAME', 'CUSTOM_CRITERIA', 'AD_UNIT_ID', "DATE"]
+        report_job["reportQuery"]["dimensions"] = dimensions
+        report_job["reportQuery"]["columns"] = custom_traffic_source_columns
+        report_job["reportQuery"]["adUnitView"] = 'HIERARCHICAL'
+        statement_builder = ad_manager.StatementBuilder(version='v202208')
+        statement_builder.Where(f"CUSTOM_TARGETING_VALUE_ID IN ({targets_ids})")
+        statement_builder.Limit(None).Offset(None)
+        report_job["reportQuery"]["statement"] = statement_builder.ToStatement()
+        report_job = self.add_dates_ranges(report_job, start_date, end_date)
+        return report_job
+
+    def get_custom_targeting_keys_ids(self, name: str) -> List:
+        all_keys = []
+        custom_targeting_service = self.google_ad_manager_client.GetService('CustomTargetingService', version='v202208')
+        statement_builder = ad_manager.StatementBuilder(version='v202208')
+        page_size = ad_manager.SUGGESTED_PAGE_LIMIT
+        if name:
+            statement_builder = statement_builder.Where("name = :name").WithBindVariable('name', name)
+        statement_builder.limit = page_size
+    
+        total_result_set_size = 0
+        while True:
+            response = custom_targeting_service.getCustomTargetingKeysByStatement(statement_builder.ToStatement())
+            if 'results' in response and len(response['results']):
+                all_keys.extend(response['results'])
+                statement_builder.offset += statement_builder.limit
+            else:
+                break
+        customs_target_values = self.get_custom_targeting_values(all_keys)
+        return customs_target_values
+
+    def get_custom_targeting_values(self, all_keys: list) -> List:
+        """given a list of keys retrieve the custom targeting values
+
+        Args:
+            all_keys (list): _description_
+
+        Returns:
+            dict: _description_
+        """
+        targeting_values = list()
+        if all_keys:
+            statement_builder = ad_manager.StatementBuilder(version='v202208')
```

Comment:
```suggestion
            statement_builder = ad_manager.StatementBuilder(version=API_VERSION)
```
and create a constant.

Reply:
cool.